### PR TITLE
HPCC-17316 Fix various windows build issues

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith windows didn't like non exported functions being defined within an exported class.  Making the members local static is likely to be marginally more efficient.
@richardkchapman the build-config.h is moved because of changes in HPCC-17199.  I'm slightly concerned that even though this now compiles, it may not work very well on windows.  Is it going to cause any other problems?

**Comment by ghalliday** [comment]
@richardkchapman note second change - should probably be regression tested to check my assertion

File: system/jlib/jptree.cpp [hunk 1]
```diff
@@ -69,26 +69,32 @@ IPropertyTreeIterator *createNullPTreeIterator() { return LINK(nullPTreeIterator
...
 
 MODULE_EXIT()
 {
     nullPTreeIterator->Release();
-    CAtomPTree::kill();
+	delete attrHT;
+	keyTable->Release();
+	keyTableNC->Release();
+	free(freelist);
+	freelist = NULL;
```
Comment (jakesmith) [inline system/jlib/jptree.cpp]: @trivial - not new, this line wasn't need before/now.

**Comment by jakesmith** [comment]
@ghalliday - looks fine.

**Pull request merged**